File: BaekJoon/Problems/Step/09/1978.cpp
// [BOJ] 단계별로 풀어보기 > 9. 기본 수학 2

// 1978. 소수 찾기

/* Sample input & output data
4
1 3 5 7

3
*/


#include <iostream>
#include <vector>

using namespace std;
#define endl '\n'

int main()
{
    // Input data
    int n;          // n <= 100
    cin >> n;
    vector<int> v;  // v_i <= 1000
    int temp;
    for (int i = 0; i < n; i++)
    {
        cin >> temp;
        v.push_back(temp);
    }

    // Determine if each number of v is a prime number
    int count = 0, prime;
    for (int j = 0; j < n; j++)
    {
        if (v[j] == 1) continue;

        prime = 1;
        for (int k = 2; k < v[j]/2 + 1; k++)
        {
            if (v[j] % k  == 0)
            {
                prime = 0;
                break;
            }
        }
        if (prime == 1) count++;

        // test
        // cout << j << " " << v[j] << " " << count << endl;
    }

    // Output
    cout << count << endl;

    return 0;
}

File: BaekJoon/Problems/Step/09/1978.rs
// 1978. 소수 찾기
// 2024.04.05


use std::io;

fn is_prime(num: i32, test: bool) -> bool
{
    if num == 1 { return false; }
    else if num == 2 { return true; }                       // crazy
    else
    {
        for i in 2..=(num as f32).sqrt().ceil() as i32
        {
            if test { println!("{} % {} = {}", num, i, num % i); }
            if num % i == 0 { return false; }
        }
        return true;
    }
}

fn main()
{
    // let test: bool = true;
    let test: bool = false;

    // Input n and n numbers
    let mut n = String::new();                              // not use
    let mut nums = String::new();
    io::stdin().read_line(&mut n).unwrap();
    io::stdin().read_line(&mut nums).unwrap();
    let nums: Vec<i32> = nums.split_whitespace()
        .map(|s| s.parse().unwrap()).collect();

    // Count the number of prime numbers
    let mut cnt: i32 = 0;
    for i in 0..nums.len()
    {
        if is_prime(nums[i], test) { cnt += 1; }
    }

    // Output
    println!("{}", cnt);
}
